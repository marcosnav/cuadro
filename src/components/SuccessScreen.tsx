import React, { FC } from 'react';
import { icons } from './../constants';
import PrettyLink from './PrettyLink';
import * as S from './styled';

interface IProps {
  author: string;
  creditsUrl: string;
  image: string;
  show?: boolean;
}

const SuccessScreen: FC<IProps> = ({ author, creditsUrl, image, show }) => {
  return (
    <S.SolvedWallpaper image={image} show={show}>
      <S.ControlIcon src={icons.check} />
      <S.MainTitle>
        Well Done!
      </S.MainTitle>
      <S.Credits>
        {'Picture by '}
        <PrettyLink href={creditsUrl}>
          {author}
        </PrettyLink>
      </S.Credits>
    </S.SolvedWallpaper>
  );
};

SuccessScreen.defaultProps = {
  show: false,
};

export default SuccessScreen;
