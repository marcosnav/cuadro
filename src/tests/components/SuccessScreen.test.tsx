import { mount } from 'enzyme';
import React from 'react';
import SuccessScreen from './../../components/SuccessScreen';

describe('<SuccessScreen />', () => {
  test('renders properly', () => {
    const component = mount(
    <SuccessScreen author={'John Doe'} image={'/cat-image.jpg'} creditsUrl={'https://john-doe.com'} />);
    expect(component.text()).toBe('Well Done!Picture by John Doe');
    expect(component).toMatchSnapshot();
  });
});
