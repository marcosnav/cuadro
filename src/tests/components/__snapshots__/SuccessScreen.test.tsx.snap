// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SuccessScreen /> renders properly 1`] = `
<SolvedWallpaper
  image="/cat-image.jpg"
  show={false}
>
  <div
    className="sc-bdVaJa sc-kAzzGY cPZpwQ"
  >
    <ButtonIcon
      src="check.svg"
    >
      <img
        className="sc-ifAKCX hRfgbl"
        src="check.svg"
      />
    </ButtonIcon>
    <MainTitle>
      <h1
        className="sc-gzVnrw hQfdgp"
      >
        Well Done!
      </h1>
    </MainTitle>
    <Credits>
      <p
        className="sc-fjdhpX dasJpR"
      >
        Picture by 
        <Link
          href="https://john-doe.com"
          onClick={[Function]}
          target="_blank"
        >
          <a
            className="sc-bZQynM fKVASX"
            href="https://john-doe.com"
            onClick={[Function]}
            target="_blank"
          >
            John Doe
          </a>
        </Link>
      </p>
    </Credits>
  </div>
</SolvedWallpaper>
`;
